lib/puppet/type/file.rb

```diff
@@ -821,13 +819,26 @@ def write(property)
       end
     else
       umask = mode ? 000 : 022
-      Puppet::Util.withumask(umask) { ::File.open(self[:path], 'wb', mode_int ) { |f| write_content(f) } }
+      Puppet::Util.withumask(umask) { ::File.open(self[:path], 'wb', mode_int ) { |f| write_contents(f) } }
     end
 
     # make sure all of the modes are actually correct
     property_fix
   end
 
+  def write_temporarily
+    tempfile = Tempfile.new("puppet-file")
+    tempfile.open
+
+    write_contents(tempfile)
+
+    tempfile.close
+
+    yield tempfile.path
+
+    tempfile.delete
```

Comment:
Might be good in maint commit to ensure this tempfile gets deleted. (I know it was that way before, but seems like it could leak temp file contents)?